apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LoggingConfiguration.java

```diff
@@ -159,8 +159,38 @@ public void assertValid(Boolean value) {
         })
         .buildWithDefault(false);
 
+    private final ConfigurationOption<Boolean> logShadingEnabled = ConfigurationOption.booleanOption()
+        .key("log_shading_enabled")
+        .configurationCategory(LOGGING_CATEGORY)
+        .description("A boolean specifying whether the agent should automatically reformat application logs \n" +
+            "into ECS-compatible JSON files, suitable for ingestion into Elasticsearch for further analysis. \n" +
+            "If true, check out additional `log_shading` configurations options.")
+        .dynamic(true)
+        .buildWithDefault(true);
+
+    private final ConfigurationOption<Boolean> logShadingReplace = ConfigurationOption.booleanOption()
+        .key("log_shading_replace")
+        .configurationCategory(LOGGING_CATEGORY)
+        .tags("performance")
+        .description("By default, when Log Shading is enabled, application logs will be duplicated so that the \n" +
+            "ECS-formatted logs are written to new files having the `.ecs.json` extension. In order to reduce the \n" +
+            "related overhead, set this option to true to replace the original log files with the ECS-compatible ones.")
+        .dynamic(false)
+        .buildWithDefault(false);
```

Comment:
This overlaps/conflicts with https://github.com/elastic/apm/issues/373 as the replacement should not only happen for file appenders but for any appenders that use a plain-text formatter.
We might have to do that at the time the formatter is set on the appender as we need the appender name to set the `event.dataset` (`${service.name}.${appender.name}`). This implies that it would not work for runtime attachment. But I think that's fair as it ensures that there's no mix of plain-text and JSON logs.

Reply:
We decided on using a new config option called `log_ecs_reformatting` that will have multiple options:
- OFF
- SHADE - writing to an additional file in ECS format
- REPLACE - reformat to ECS like SHADE, but block writing the original log file
- OVERRIDE - use the same logger output stream, but override with ECS formatting (the option you are referring to, which will be relevant to non-file-appenders as well)

This way they will not overlap or conflict.